modules/core/src/main/scala/parser.scala

```diff
@@ -210,29 +212,29 @@ object GraphQLParser {
 
   lazy val Selection: Parser[Ast.Selection] =
     Field.widen[Ast.Selection] |
-    FragmentSpread.widen       |
+    FragmentSpread.widen.backtrack       |
```

Comment:
can we possibly get rid of backtrack or comment what kinds of expressions need it? Maybe as easy as reversing the order?

Reply:
The issue is that multiple things can be prefixed by a string description so if we parse that then we need to backtrack. I had some work in progress to restructure the AST so we parse the comment separately and then attach it to the selection so I'll carry on with that 👍